sdk/onlineexperimentation/azure-analytics-onlineexperimentation/src/main/java/com/azure/analytics/onlineexperimentation/OnlineExperimentationClient.java

```diff
@@ -0,0 +1,694 @@
+     * @param resource The resource instance.
+     * @param ifMatch The ETag value to match for the request.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
+     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return defines experiment metric metadata and computation details.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public ExperimentMetric updateMetric(String experimentMetricId, ExperimentMetric resource, String ifMatch) {
+        return createOrUpdateMetric(experimentMetricId, resource, new RequestConditions().setIfMatch(ifMatch));
+    }
+
+    /**
+     * Creates or updates an experiment metric.
+     *
+     * @param experimentMetricId Identifier for this experiment metric. Must start with a lowercase letter and contain
+     * only lowercase letters, numbers, and underscores.
+     * @param resource The resource instance.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
+     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return defines experiment metric metadata and computation details.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public ExperimentMetric updateMetric(String experimentMetricId, ExperimentMetric resource) {
+        return updateMetric(experimentMetricId, resource, ETag.ALL.toString());
+    }
+
+    /**
+     * Updates the experiment metric lifecycle to {@link LifecycleStage#ACTIVE}.
+     *
+     * @param experimentMetricId Identifier for this experiment metric. Must start with a lowercase letter and contain
+     * only lowercase letters, numbers, and underscores.
+     * @param ifMatch The ETag value to match for the request.
+     *
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return defines experiment metric metadata and computation details.
+     */
+    public ExperimentMetric activateMetric(String experimentMetricId, String ifMatch) {
```

Comment:
They seems still be `@ServiceMethod`, as it calls the REST API?

Reply:
Good catch. The async client methods had annotations for the non-generated methods but sync client was missing them.